textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,734 @@
+                        .append(id, statement.id)
+                        .isEquals();
+        }
+    }
+    static public class SelectStatement extends Statement {
+        /**
+         * Set to true when '*' is used as the fields to be projected, as in "SELECT * FROM ..."
+         */
+        public Boolean projectAll;
+        
+        /**
+         * The { @link List } of fields to be projected if it is specified as
+         * in "SELECT a, b, c FROM ..."
+         */
+        public List<String> projectedFields;
+        
+        
+        /**
+         * Predicate for used for data extraction such as keyword match in "KEYWORDMATCH(a,"word")"
+         */
+        public ExtractPredicate extractPredicate;
+
+        /**
+         * The identifier of a view or a table name, as in "SELECT... FROM viewName"
+         * used as source of tuples
+         */
+        public String fromClause;
+
+        /**
+         * The maximum number of tuples to be returned, as in "SELECT...FROM... LIMIT 5"
+         */
+        public Integer limitClause;
+
+        /**
+         * The number of tuples to be skipped before returning, as in "SELECT...FROM... OFFSET 5"
+         */
+        public Integer offsetClause;
+
+        public SelectStatement() {
+            this(null, null, null, null, null, null, null);
+        }
+        public SelectStatement(String id, Boolean projectAll,
+                              List<String> projectedFields, ExtractPredicate extractPredicate,
+                              String fromClause, Integer limitClause, Integer offsetClause) {
+            super(id);
+            this.projectAll = projectAll;
+            this.projectedFields = projectedFields;
+            this.extractPredicate = extractPredicate;
+            this.fromClause = fromClause;
+            this.limitClause = limitClause;
+            this.offsetClause = offsetClause;
+        }
+        @Override
+        public boolean equals(Object other) {
+            if (other == null) { return false; }
+            if (other.getClass() != this.getClass()) { return false; }
+            SelectStatement selectStatement = (SelectStatement) other;
+            return new EqualsBuilder()
+                        .appendSuper(super.equals(selectStatement))
+                        .append(projectAll, selectStatement.projectAll)
+                        .append(projectedFields, selectStatement.projectedFields)
+                        .append(extractPredicate, selectStatement.extractPredicate)
+                        .append(fromClause, selectStatement.fromClause)
+                        .append(limitClause, selectStatement.limitClause)
+                        .append(offsetClause, selectStatement.offsetClause)
+                        .isEquals();
+        }
+    }
+    static public class CreateViewStatement extends Statement {
+        /**
+         * The { @link Statement } to which the create view statement creates an alias for,
```

Comment:
"to which the create view statement creates an alias for" -> "for which the create view statement creates an alias,"